toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -185,13 +313,206 @@ func resolveGraphNodes(dependencyGraph *pkggraph.PkgGraph, inputSummaryFile, out
-		err = repoutils.SaveClonedRepoContents(cloner, outputSummaryFile)
+	// Fix up the implicit nodes to point to the correct delta RPMs that we parsed earlier.
+	for _, n := range skippedNodes {
+		logger.Log.Debugf("Fixing up skipped node %s", n)
+		err = fixupDeltaImplicitNode(realDependencyGraph, n, &originalPathToDeltaPathMap)
 		if err != nil {
-			logger.Log.Errorf("Failed to save cloned repo contents.")
-			return
+			return fmt.Errorf("failed to fixup skipped node %s: %w", n, err)
+		}
+	}
+
+	logger.Log.Info("Configuring additional delta RPMs as a local repository")
+	err = cloner.ConvertDownloadedPackagesIntoRepo()
+	if err != nil {
+		logger.Log.Errorf("Failed to convert downloaded RPMs into a repo. Error: %s", err)
+		return
+	}
+
+	return
+}
+
+// downloadSingleDeltaRPM attempts to download a single delta RPM for a build node. If the delta RPM is available
+// it will be downloaded and the build node will be updated to point to the new RPM. The associated run node will
+// also be updated to point to the new RPM since the scheduler uses the run node to find the RPM to install.
+//   - realDependencyGraph: The graph to update
+//   - realBuildNode: The build node to update. This node should be from the real graph as we will be updating it directly.
+//     to find the actual build node in the graph.
+//   - cloner: The cloner to use to download the RPMs
+//   - deltaRpmDir: The directory to download the RPMs into (likely the same as the normal RPM cache)
+//   - pathMap: A map of the original path to the delta path for each RPM that was downloaded. This is used to fix up
+//     the implicit nodes later.
+func downloadSingleDeltaRPM(realDependencyGraph *pkggraph.PkgGraph, realBuildNode *pkggraph.PkgNode, cloner *rpmrepocloner.RpmRepoCloner, deltaRpmDir string, pathMap *map[string]string) (foundMatch bool, err error) {
+	//TODO: downloadDependencies is broken right now, fix it
+	const downloadDependencies = true
+	var lookup *pkggraph.LookupNode
+
+	// Find the real build node in the graph we want to keep (we will be discarding the graph the node was passed in from so we can't use it)
+	if realBuildNode.Type != pkggraph.TypeBuild {
+		err = fmt.Errorf("node '%s' is not a build node, can't download delta RPM", realBuildNode)
+		return false, err
+	}
+
+	timestamp.StartEvent(fmt.Sprintf("downloading delta node %s", realBuildNode.VersionedPkg.Name), nil)
+	defer timestamp.StopEvent(nil)
+
+	lookup, err = realDependencyGraph.FindExactPkgNodeFromPkg(realBuildNode.VersionedPkg)
+	if err != nil {
+		err = fmt.Errorf("can't find build node '%s' in graph: %w", realBuildNode, err)
+		return false, err
+	}
+	if lookup == nil || lookup.RunNode == nil {
+		err = fmt.Errorf("can't find run lookup '%v' in graph", lookup)
+		return false, err
+	}
+
+	if lookup.BuildNode != realBuildNode {
+		err = fmt.Errorf("real build node '%v' does not match build node in the graph lookup '%v'", realBuildNode, lookup.BuildNode)
+		return false, err
+	}
```

Comment:
Curious: wouldn't we be able to drop `FindExactPkgNodeFromPkg` along with the checks we perform below it if we instead looked through the nodes pointing to `realBuildNode` and grabbed the run node from them? I think we might always have only one node pointing to a build node at the moment, since we create a run -> build edge for each pair of packages and provides coming from a single spec.

Reply:
I think we have no hard requirement for that due to cycle resolution meta nodes and the like. In the simple case this is true but I'd rather not try to figure out how the graph has been modified.

Reply:
Fair enough.